server/src/main/java/io/spine/server/event/EEntity.java

```diff
@@ -162,7 +162,8 @@ private static Event compact(Event event) {
                 // Does nothing.
                 break;
             case ORIGIN_NOT_SET:
-                throw newIllegalStateException("EventContext.origin should be set.");
+                // Does nothing because the origins is `null`.
+                break;
             default:
                 throw newIllegalStateException("Not all of `OriginCase` values were handled.");
```

Comment:
Have the message which says something like “Unsupported origin case encountered: %s”.